packages/contracts/contracts/plugin/PluginRepo.sol

```diff
@@ -0,0 +1,233 @@
+        versionIdxForSemantic[semanticVersionHash(_newSemanticVersion)] = versionIdx;
+        versionIdxForContract[_pluginFactoryAddress] = versionIdx;
+
+        emit NewVersion(versionIdx, _newSemanticVersion);
+    }
+
+    /// @notice get latest plugin
+    /// @return semanticVersion Semantic version for latest pluginRepo version
+    /// @return pluginFactoryAddress Address of latest plugin factory for version
+    /// @return contentURI External URI for fetching latest version's content
+    function getLatest()
+        public
+        view
+        returns (
+            uint16[3] memory semanticVersion,
+            address pluginFactoryAddress,
+            bytes memory contentURI
+        )
+    {
+        return getByVersionId(nextVersionIndex - 1);
+    }
+
+    /// @notice get latest by plugin factory address
+    /// @return semanticVersion Semantic version for pluginRepo version
+    /// @return pluginFactoryAddress Address of plugin factory for version
+    /// @return contentURI External URI for fetching version's content
+    function getLatestForContractAddress(address _pluginFactoryAddress)
+        public
+        view
+        returns (
+            uint16[3] memory semanticVersion,
+            address pluginFactoryAddress,
+            bytes memory contentURI
+        )
+    {
+        return getByVersionId(versionIdxForContract[_pluginFactoryAddress]);
+    }
+
+    /// @notice get latest by semantic version
+    /// @return semanticVersion Semantic version for latest pluginRepo version
+    /// @return pluginFactoryAddress Address of plugin factory for version
+    /// @return contentURI External URI for fetching latest version's content
+    function getBySemanticVersion(uint16[3] memory _semanticVersion)
+        public
+        view
+        returns (
+            uint16[3] memory semanticVersion,
+            address pluginFactoryAddress,
+            bytes memory contentURI
+        )
+    {
+        return getByVersionId(versionIdxForSemantic[semanticVersionHash(_semanticVersion)]);
+    }
+
+    /// @notice get latest by version id
+    /// @return semanticVersion Semantic version for pluginRepo version
+    /// @return pluginFactoryAddress Address of plugin factory for version
+    /// @return contentURI External URI for fetching version's content
+    function getByVersionId(uint256 _versionIdx)
+        public
+        view
+        returns (
+            uint16[3] memory semanticVersion,
+            address pluginFactoryAddress,
+            bytes memory contentURI
+        )
+    {
+        if (_versionIdx <= 0 || _versionIdx >= nextVersionIndex) revert VersionDoesNotExist();
+        Version storage version = versions[_versionIdx];
+        return (version.semanticVersion, version.pluginFactoryAddress, version.contentURI);
+    }
+
+    /// @notice get version count
+    /// @return uint256 count value
+    function getVersionCount() public view returns (uint256) {
+        return nextVersionIndex - 1;
```

Comment:
this will be off by 1 because the `nextVersionIndex` starts at 0. remove the `-1`

Reply:
in the initialize function `nextVersionIndex` is set to 1
so it will not be off,
do u agree?

Reply:
couldn't you then just set the default value to 1 and save the gas in the initialize function?

Reply:
while this might work if we deploy directly this contract.
but it won't work if we deploy this contract via proxy